apis/v1beta1/validation/httproute_test.go

```diff
@@ -1176,3 +1176,67 @@ func TestValidateRequestRedirectFiltersWithNoBackendRef(t *testing.T) {
 		})
 	}
 }
+
+func toDuration(durationString string) *gatewayv1b1.Duration {
+	return (*gatewayv1b1.Duration)(&durationString)
+}
+
+func TestValidateHTTPTimeouts(t *testing.T) {
+	tests := []struct {
+		name     string
+		rules    []gatewayv1b1.HTTPRouteRule
+		errCount int
+	}{
+		{
+			name:     "valid httpRoute Rules timeouts",
+			errCount: 0,
+			rules: []gatewayv1b1.HTTPRouteRule{
+				{
+					Timeouts: &gatewayv1b1.HTTPRouteTimeouts{
+						Request: toDuration("1ms"),
+					},
+				},
+			},
+		}, {
+			name:     "valid httpRoute Rules timeout set to 0 (disabled)",
+			errCount: 0,
+			rules: []gatewayv1b1.HTTPRouteRule{
+				{
+					Timeouts: &gatewayv1b1.HTTPRouteTimeouts{
+						Request: toDuration("0ms"),
```

Comment:
Although "0ms" also works, the spec recommends using "0s" for 0 (infinite) timeout, so best to follow best practices here?
```suggestion
						Request: toDuration("0s"),
```

Reply:
addressed in 6281ef19931f78d873d920088ba4abb36055f346